File: chapter-3-greedy/거스름돈-book.py
import time


def solution(N):
    count = 0
    coin_types = [500,100,50,10]

    for coin in coin_types:
        count += N // coin # 해당 화폐로 거슬러 줄 수 있는 동전의 개수 세기
        N %= coin

    return count


N = 1260
start_time = time.time()
result = solution(N)
end_time = time.time()

print('결과:',result)
print(f'실행시간: {end_time-start_time}')

File: chapter-3-greedy/거스름돈-book.cpp
#include <iostream>

using namespace std;

int solution(int N){
    int count = 0;
    int coin_types[4] = {500, 100, 50, 10};
    int len = sizeof(coin_types)/sizeof(int);

    for(const auto& coin : coin_types){
        count += N / coin;
        N %= coin;
    }

    return count;
} 

int main(){
    int N = 1260;
    int result = solution(N);
    cout << "결과: " << result << endl;
    return 0;
}
